scripts/pre-push-gate.sh

```
@@ -248,5 +248,39 @@ else
 fi
 
+echo ""
+echo "=== Per-package coverage floor ==="
+# Enforce the one-way coverage ratchet. Each internal/ package must stay at
+# or above the floor recorded in testdata/coverage-floor.json. Reuses the
+# coverage profile generated by the test step above ($COVER_OUT) so no
+# second test run is needed.
+#
+# coverage-floor.sh uses exit codes 0|1|2. Collapse non-zero to 1 here,
+# consistent with how patch-coverage.sh exits are handled above.
+if COVER_OUT="$COVER_OUT" bash "$SCRIPT_DIR/coverage-floor.sh" --cover "$COVER_OUT"; then
```

Shellcheck (0.11.0)

[warning] This assignment is only seen by the forked process.

(SC2097)